src/backend/BackendInterface.h

```diff
@@ -37,29 +37,52 @@ retryOnTimeout(F func, size_t waitMs = 500)
     }
 }
 
-// Please note, this function only works w/ non-void return type. Writes are
-// synchronous anyways, so
 template <class F>
-void
+auto
 synchronous(F&& f)
 {
     boost::asio::io_context ctx;
+    boost::asio::io_context::strand strand(ctx);
     std::optional<boost::asio::io_context::work> work;
 
     work.emplace(ctx);
 
-    boost::asio::spawn(ctx, [&f, &work](boost::asio::yield_context yield) {
-        f(yield);
+    using R = typename std::result_of<F(boost::asio::yield_context)>::type;
+    if constexpr (!std::is_same<R, void>::value)
```

Comment:
So that's how you do that...

I spent a good chunk of time trying to figure this one out.